### ISSUE TITLE
a11y: Fix missing role attribute issue with switch control

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12817]
a11y issue: 207

**Issue 207 - Switch control not programmatically determined**

The role such as "Switch" is not programmatically determined which resulted in screen reader users not being able to interact with it. Additionally, when user activates it "checked"/"not checked" state is announcing for screen reader users.

Occurrences:

Users Retention Settings
Main Content
- Run the user retention process in DRY mode (no changes will be applied (available on activating "Disable user accounts after an inactivity period (duration since last login)" and "Delete user accounts after an inactivity period (duration since last login)" checkboxes.

Cluster Management - Create
- RKE2/K3s

Example:

Users Retention Settings
Main Content
- Run the user retention process in DRY mode (no changes will be applied (available on activating "Disable user accounts after an inactivity period (duration since last login)" and "Delete user accounts after an inactivity period (duration since last login)" checkboxes.

Cluster Management - Create
- RKE2/K3s

Suggest remediation:

Apply the following changes:
• Provide role="switch" to <label> element containing switch control.
• Provide "aria-checked" attribute to <label> element containing switch control. Ensure that the value of "aria-checked" attribute gets updated on user interaction via scripting.

A working example of custom switch control can be found at:https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch/

Alternatively, 
Replace the custom checkbox with the standard checkbox by using the <input type="checkbox"> element. If required use CSS to maintain the presentation of the page.

Alternatively, use appropriate ARIA roles and attributes to code the custom checkbox. 

A working example of custom checkbox can be found at:
https://www.w3.org/WAI/ARIA/apg/example-index/checkbox/checkbox.html

**Comment by yonasberhe23** [comment]
Requires manual validation.

**Comment by izaac** [comment]
Rancher version: `v2.11-0b081e077e830fa17b86151d21c786fb8d07c6de-head`
UI version: `master 68fb044`
Chrome version: `133.0.6943.141`
Firefox version: `135.0.1`

I validated the areas described in this issue. The screen reader picked the text properly and I am able to navigate to all elemwnts with only the keyboard.
